### PR TITLE
Feature/kak/fix chart year after nav#242

### LINKED ISSUE #242: Incorrect charts header year after navigation
If the year is changed while on a different tab, then the tab changes to 'charts', the report year displayed in the header is incorrect. Changing the year directly on the 'charts' tab updates correctly. Header report year should be selected year + 1.

Here, the year 2013 has been picked from the drop-down, but the report year is given as 2015. It should be 2014.
image

Verify that the correct year is used for the report data after navigation as well.
See discussion on #241.

Issue has been present for some time, as it also occurs on the [production site](http://visualization.phillybuildingbenchmarking.com/#/).

### PR DESCRIPTION
**Pull request by flibbertigibbet**
Consistently keep `year` parameter in URL. The `getCurrentYear` function reads the selected year from the URL, or else falls back to a default. On a change to the year picker, the year gets set in the URL. However, the `year` parameter was getting dropped from the URL on state changes, resulting in potential mismatch between selected and displayed years.

Passing the parameter required refactoring to move the header bar from the `index.html` file into an angular component, so the `ui-sref` links for the map and charts pages could reference state.

Fixes #242

### CODE REVIEW

File: app/scripts/headerbar/headerbar-partial.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<div>
+    <a ui-sref="charts"><div id="logo"></div></a>
+    <ul class="nav nav-pills pull-right">
+
+        <mos-year-selector></mos-year-selector>
+
+        <li ui-sref-active="active"><a ui-sref="charts({year: ctl.year})"><div class="nav-icon nav-chart"></div>Charts</a></li>
+        <!-- TODO: Figure out how to show map highlight when detail/compare is active? -->
```
Comment (CloudNiner) [inline app/scripts/headerbar/headerbar-partial.html]: An old TODO! Looks like this was copied from the old HTML. Is it still valid?

Comment (flibbertigibbet) [reply app/scripts/headerbar/headerbar-partial.html]: I don't even know what it means. Shall I make a mystery issue?

**Comment by flibbertigibbet** [comment]
I didn't update the `compare` and `detail` views since they display data for all of the years, although now you mention it, I do see there's some summary values that must be for the current year. I'll change those too.

**Comment by flibbertigibbet** [comment]
Rooting around in the `detail` view I found [this note](https://github.com/azavea/mos-energy-benchmark/blob/develop/app/scripts/views/detail/detail-controller.js#L36) saying the year toggle is intentionally ignored there, but there's no reason given as to why.

**Comment by flibbertigibbet** [comment]
Updated for `compare` and `detail` pages. There was a note that the `detail` intentionally ignores the `year` but updated it as well, for consistency.

**Pull request merged**